### PR TITLE
[WIP] Add tests for DOM methods

### LINKED ISSUE #4252: Unit tests missing for some DOM methods
#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [x] Other (DOM)

#### Feature enhancement details:
We need to add unit tests for several methods of DOM.
Unit tests missing for methods like createP(), createDiv(), createImg(),createA(),createSlider(),createRadio(),etc.

### PR DESCRIPTION
**Pull request by jatin33**
Resolves #4252 

 Changes: 

Adding tests for missing DOM methods
- [x] createSlider()
- [x] createA
- [ ] createCheckBox()
- [ ] createRadio()

#### PR Checklist

- [x] `npm run lint` passes
- [ ] [Inline documentation] is included / updated
- [x] [Unit tests] are included / updated
- [ ] [Benchmarks] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/master/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/master/contributor_docs#unit-tests
[Benchmarks]: https://github.com/processing/p5.js/blob/master/contributor_docs/benchmarking_p5.md

### CODE REVIEW

**Comment by jatin33** [comment]
Still Work in progress, going to add tests for more methods.

**Comment by lmccart** [comment]
nice approach. looks good so far

**Comment by jatin33** [comment]
Hey @lmccart , please take a look at 48471cc , here I have trouble understanding testSketchWithPromise() and I went through this method definition [here](https://github.com/processing/p5.js/blob/master/test/js/p5_helpers.js#L17) and understood that it returns a promise hence attached a then block. Is this the correct approach, please guide me if I am wrong.

**Comment by jatin33** [comment]
Also, I have edited the PR details, adding which methods testing I would be covering.

File: test/unit/dom/dom.js [hunk 1]
```diff
@@ -222,4 +222,105 @@ suite('DOM', function() {
...
+    });
+
+    testSketchWithPromise('anchor tag is hovered', function(
+      sketch,
+      resolve,
+      reject
+    ) {
+      sketch.setup = function() {
+        var elem = sketch.createA('http://p5js.org/', 'this is a link');
+        elem.mouseMoved();
```
Comment (lmccart) [inline test/unit/dom/dom.js]: I'm not sure I understand this test, can you explain?
it looks like the `testSketchWithPromise` is used correctly in any case

Comment (jatin33) [reply test/unit/dom/dom.js]: I was confused as it was used a bit differently [here](https://github.com/processing/p5.js/blob/master/test/unit/dom/dom.js#L172) as there was no then block used for resolution.

**Pull request merged**